test/unit_tests/braket/experimental/autoqasm/test_api.py

```diff
@@ -792,3 +792,26 @@ def main() -> None:
 
     with pytest.raises(errors.InconsistentNumQubits):
         main()
+
+
+def test_nested_function():
+    @aq.function
+    def make_ghz(n: int) -> None:
+        def ghz(n: int):
+            if n == 1:
+                h(0)
+            else:
+                ghz(n - 1)
+                cnot(0, n - 1)
+
+        ghz(n)
+
+    expected = """OPENQASM 3.0;
```

Comment:
cool test

Reply:
You can thank @godott for this test (which uncovered the bug), it's copied directly from his bug report.